src/network/utils.ts

```diff
@@ -215,3 +215,11 @@ export function validateMultiaddrs(multiaddrs) {
   }
   return true;
 }
+
+export function validateTimestamp(timestamp, validPeriod) {
+  let diff = Date.now() - timestamp;
+  if (diff < 0)
+    throw `Discovery timestamp cannot be future time`;
```

Comment:
Why `Discovery timestamp`? If it is a general function, it could be used anywhere and the errs should be general.